src/frontend/mame/infoxml.cpp

```diff
@@ -1809,6 +1815,40 @@ void output_driver(std::ostream &out, game_driver const &driver, device_t::featu
 	else
 		out << " savestate=\"unsupported\"";
 
+	if (flags & machine_flags::TYPE_ARCADE)
+		out << " type=\"arcade\"";
+	else if (flags & machine_flags::TYPE_CONSOLE)
+		out << " type=\"console\"";
+	else if (flags & machine_flags::TYPE_COMPUTER)
+		out << " type=\"computer\"";
+	else if (flags & machine_flags::TYPE_OTHER)
+		out << " type=\"other\"";
+
+	if (flags & machine_flags::REQUIRES_ARTWORK)
+		out << " requiresartwork=\"yes\"";
+	else
+		out << " requiresartwork=\"no\"";
+
+	if (flags & machine_flags::CLICKABLE_ARTWORK)
+		out << " clickableartwork=\"yes\"";
+	else
+		out << " clickableartwork=\"no\"";
```

Comment:
This is also a poorly-considered flag that should really be an attribute of the view in the artwork, not a flag on the machine.  It’s going to be removed along with other artwork system updates.